addons/godot-xr-tools/objects/grab_points/grab_point_hand.gd

```diff
@@ -0,0 +1,131 @@
+tool
+class_name XRToolsGrabPointHand
+extends XRToolsGrabPoint
+
+
+## XR Tools Grab Point Hand Script
+##
+## This script allows specifying a grab point for a specific hand. Additionally
+## the grab point can be used to control the pose of the hand, and to allow the
+## grab point position to be fine-tuned in the editor.
+
+
+## Left hand scene path (for editorpreview)
+const LEFT_HAND_PATH := "res://addons/godot-xr-tools/hands/scenes/lowpoly/left_hand_low.tscn"
+
+## Right hand scene path (for editor preview)
+const RIGHT_HAND_PATH := "res://addons/godot-xr-tools/hands/scenes/lowpoly/right_hand_low.tscn"
+
+
+## Hand for this grab point
+enum Hand {
+	LEFT,	## Left hand
+	RIGHT	## Right hand
+}
+
+## Hand preview option
+enum PreviewMode {
+	CLOSED	## Preview hand closed
+	OPEN,	## Preview hand open
+}
+
+
+## Which hand this grab point is for
+export (Hand) var hand : int setget _set_hand
+
+## Hand pose
+export var hand_pose : Resource setget _set_hand_pose
+
+## If true, the hand is shown in the editor
+export (PreviewMode) var editor_preview_mode : int = PreviewMode.CLOSED setget _set_editor_preview_mode
+
+
+## Hand to use for editor preview
+var _editor_preview_hand : XRToolsHand
+
+
+## Called when the node enters the scene tree for the first time.
+func _ready():
+	# If in the editor then update the preview
+	if Engine.editor_hint:
+		_update_editor_preview()
+
+
+## Test if a grabber can grab by this grab-point
+func can_grab(_grabber : Node) -> bool:
+	# Skip if not enabled
+	if not enabled:
+		return false
+
+	# Ensure the pickup is valid
+	if not is_instance_valid(_grabber):
+		return false
+
+	# Ensure the pickup is a function pickup for a controller
+	var pickup := _grabber as XRToolsFunctionPickup
+	if not pickup:
+		return false
+
+	# Get the parent controller
+	var controller := _grabber.get_parent() as ARVRController
```

Comment:
I think we should add a `get_controller` function to `XRToolsFunctionPickup` to future proof this if we ever decide to make it possible to add the pickup function to another node.

Reply:
Added the get_controller() to XRToolsFunctionPickup, and added its use in XRToolsGrabPointHand.